### DISCUSSION TITLE
Nested application Structure under apps

### DISCUSSION THREAD

**Discussion by anishmiviewis** [nrwl/nx#19370] [General]
I have question regarding if we can have nested app structure under apps folder in nx like below ?

libs/
apps/web/web-app1
apps/web/web-app2
apps/mobile/mobile-app1
apps/mobile/ mobile-app2

will this structure create any problem in the future nx upgrades and such?

**Comment by anishmiviewis** [comment]
Hi @FrozenPandaz Please accept my apologies for mentioning you here. But Can you please help me clarify above question please.

**Reply by anishmiviewis** [reply]
Hi @xiongemi Please accept my apologies for mentioning you here. But our team was awaiting on this for changing our folder structure. Can you please help us clarify this question please

**Comment by moatorres** [comment]
Hey @anishmiviewis, you can have nested folders, no problems. 👍 Here's a [live example](https://github.com/codyslexia/nexa/tree/main/apps/graphql) made by me. You can see that `apps/graphql/users` is a nested directory where `users` is the actual project. There's also [this other example](https://github.com/nrwl/nx-incremental-large-repo/tree/master/libs/app0/lib1) from the `nrwl` family.

**Reply by anishmiviewis** [reply]
Hi @moatorres. Thank you very much for the response and example. So that mean there will not be any issues with nx upgrades with these nested folders under apps right? Just want to make sure as the docs does not mentions it explicitly. We are awaiting on the confirmation from NX team to restructure our app

**Comment by anishmiviewis** [comment]
Hi @jaysoo Please accept my apologies for mentioning you here. But our team was awaiting on this for changing our folder structure. Can you please help us clarify that will this nested folder structure be a problem in future nx upgrades? Thank you
